### PR TITLE
Remove guage & add Observer metric instrument

### LINKED ISSUE #497: Replace gauge with observer metric instrument
Implements spec change [430](https://github.com/open-telemetry/opentelemetry-specification/pull/430).

- Removes gauge metric
- Add observer (includes adding new LastValueAggregator)

### PR DESCRIPTION
**Pull request by MikeGoldsmith**
Fixes #497.

### CODE REVIEW

File: src/OpenTelemetry.Api/Metrics/Meter.cs [hunk 1]
```diff
@@ -71,6 +55,22 @@ public abstract class Meter
         /// <returns>The measure instance.</returns>
         public abstract Measure<double> CreateDoubleMeasure(string name, bool absolute = true);
 
+        /// <summary>
+        /// Creates Int64 Observer with given name.
+        /// </summary>
+        /// <param name="name">The name of the observer.</param>
+        /// <param name="absolute">indicates if only positive values are expected.</param>
+        /// <returns>The observer instance.</returns>
+        public abstract Observer<long> CreateInt64Observer(string name, bool absolute = true);
```
Comment (SergeyKanzhelev) [inline src/OpenTelemetry.Api/Metrics/Meter.cs]: looks like a name clash with the https://docs.microsoft.com/en-us/dotnet/standard/events/how-to-implement-an-observer

Maybe `MetricsObserver`?

File: test/OpenTelemetry.Tests/Impl/Metrics/MetricsTest.cs [hunk 2]
```diff
@@ -55,39 +55,6 @@ public void CounterSendsAggregateToRegisteredProcessor()
             Assert.Equal(210, testProcessor.counters[0].Item3);
         }
 
-        [Fact(Skip = "TODO: Make robust tests. These are just basic tests to see metric in action.")]
```
Comment (SergeyKanzhelev) [inline test/OpenTelemetry.Tests/Impl/Metrics/MetricsTest.cs]: please add a test for observer?

Comment (MikeGoldsmith) [reply test/OpenTelemetry.Tests/Impl/Metrics/MetricsTest.cs]: Added test for Observer, and fixed/enabled the tests for measure & counter.

File: src/OpenTelemetry.Api/Metrics/Meter.cs (hunk 1 above)
Comment (MikeGoldsmith) [reply src/OpenTelemetry.Api/Metrics/Meter.cs]: I understand the concern and agree it's best to disambiguate. However, if we add a prefix, we should apply it consistently and add to all other similar types. eg `MetricsCounter`, `MetricsMeasure`, `MetricsCounterHandle`, etc.

What do you think?

Comment (MikeGoldsmith) [reply src/OpenTelemetry.Api/Metrics/Meter.cs]: A suffix would work: `ObserverMetric`, `ObserverMetricHandle`, etc

**Comment by MikeGoldsmith** [comment]
Interestingly, I'm seeing intermittent failures for the three metric tests locally. MeterSdk uses a ConcurrentDictionary to manage each metric type, using LabelSetSdk as the key which overrides GetHashCode to use a constructed string from the labels (eg label1=value1,label2=value2). The failure indicates the oder of the metric entries to be at fault.

**Comment by MikeGoldsmith** [comment]
PS the tests were previously ignored.

**Comment by MikeGoldsmith** [comment]
Figured out the problem - the assert statements are using indexes into the dictionary but it's not sorted. This means the order is not guranteed based on the `GetHashCode` value.

Working on refactoring all three metric types' tests to not depend on order.

**Comment by MikeGoldsmith** [comment]
The unit tests are now working as expected.

File: src/OpenTelemetry.Api/Metrics/Meter.cs (hunk 1 above)
Comment (MikeGoldsmith) [reply src/OpenTelemetry.Api/Metrics/Meter.cs]: Applied `Metric` suffix to counter, measure and observer

**Comment by MikeGoldsmith** [comment]
@SergeyKanzhelev @cijothomas Ready for review :+1:

**Pull request merged**